parse_str_test.go

```diff
@@ -4,27 +4,45 @@ import (
 	"fmt"
 	"reflect"
 	"testing"
+
+	"github.com/elliotchance/orderedmap/v2"
 )
 
+type dict map[any]any
+
+func convertOrderedMapToMap(omap orderedmap.OrderedMap[any, any]) dict {
+	// Convert OrderedMap to map
+	comparison := make(dict, omap.Len())
+	for el := omap.Front(); el != nil; el = el.Next() {
+		switch v := el.Value.(type) {
+		case orderedmap.OrderedMap[any, any]:
+			comparison[el.Key] = convertOrderedMapToMap(v)
+		default:
+			comparison[el.Key] = el.Value
+		}
+	}
+	return comparison
+}
```

Comment:
The inherent complexity and the unique structure of OrderedMap make it challenging to directly compare the outcomes tests against expected results. Therefore, I've added a new function, `convertOrderedMapToMap`, to facilitate comparison between OrderedMap types and standard Go maps in our tests.